File: Python/BinToDec/bin_to_dec.py
#! /usr/bin/env python3
# This program converts a binary number to a decimal number

# Define variables
binary_number = 0
decimal_number = 0
temp_number = ""
digit = 0
divisor = 1
remainder = 0
multiplier = 1

binary_number = int( input( "Enter a binary number: " ) ) # Prompt user for input

temp_number = str( binary_number )[::-1] # Assign the reverse value of binary_number to temp_number

# Use a loop to find the value of divisor
for character in temp_number:
    divisor *= 10

divisor = int( divisor / 10 ) # Assign divisor to the integer result of divisor divided by 10

# Use a loop to separate the digits in temp_number and convert them to decimal
for character in temp_number:
    digit = int( temp_number ) // divisor
    remainder = int( temp_number ) % divisor
    temp_number = remainder
    divisor /= 10

    decimal_number += digit * multiplier # add digit times multiplier to decimal_number
    multiplier *= 2 # Multiply multiplier by by

print( "{} to decimal is {}".format( binary_number, decimal_number ) )

File: C++/BinToDec/binToDec.cpp
// This program converts a binary number to a decimal number
#include <iostream>

// Function main begins program execution
int main()
{
    // Variable declarations
    int binaryNumber;
    int decimalNumber = 0;
    int remainder = 0;
    int tempNumber;
    int reverseNumber = 0;
    int divisor = 1;
    int digit;
    int multiplier = 1;

    std::cout << "Enter a binary number: "; // Prompt user for input
    std::cin >> binaryNumber; // Read binaryNumber from the user;

    tempNumber = binaryNumber; // Assign binaryNumber to tempNumber

    // While tempNumber is not zero
    while ( tempNumber != 0 )
    {
        // Find the reverse of binaryNumber
        remainder = tempNumber % 10;
        reverseNumber = reverseNumber * 10 + remainder;
        tempNumber /= 10;
    } // end while

    std::cout << "Reverse " << reverseNumber << std::endl;

    tempNumber = binaryNumber; // Assign binaryNumber to tempNumber

    // While tempNumber is not zero
    while ( tempNumber != 0 )
    {
        divisor *= 10; // Multiply the value of divisor by 10
        tempNumber /= 10; // Divide the value of tempNumber by 10
    } // end while

    divisor /= 10; // Divide the value obtained for divisor by 10 to get the correct value

    // While reverseNumber is not 0
    while ( reverseNumber != 0 )
    {
        // Separate the digits of reverseNumber
        digit = reverseNumber / divisor;
        remainder = reverseNumber % divisor;
        reverseNumber = remainder;
        divisor /= 10;

        // Convert each digit to a decimalNumber
        decimalNumber += digit * multiplier;
        multiplier *= 2;
    } // end while

    // Print the result
    std::cout << binaryNumber << " to decimal is " << decimalNumber << std::endl;

    return 0; // Indicate the program ended successfully
} // end function
